pkg/image/source.go

```diff
@@ -105,9 +113,21 @@ func detectSource(fs afero.Fs, userInput string) (Source, string, error) {
 			return UnknownSource, "", fmt.Errorf("unable to expand potential home dir expression: %w", err)
 		}
 	case UnknownSource:
-		if isDockerReference(userInput) {
-			// ignore any source hint since the source is ultimately unknown, see if this could be a docker image
-			return DockerDaemonSource, userInput, nil
+		// ignore any source hint since the source is ultimately unknown, see if this could be a docker image
+		if isRegistryReference(userInput) {
+			// verify that the docker daemon is accessible before assuming we can suggest to use it
+			dockerClient, err := docker.GetClient()
+			if err == nil {
+				ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+				defer cancel()
+				pong, err := dockerClient.Ping(ctx)
+				if err == nil && pong.APIVersion != "" {
```

Comment:
is it possible to have a registry that doesn't return an APIVersion?

Reply:
this is specific to the docker daemon, not directly to a registry --to answer your question, no, docker should always return a version if you get a ping cmd response. (context, the reason why this was changed was to allow for registry to be a fallback in case the docker daemon wasn't available... however, in order to determine that you need to reach out to the docker daemon... the "simplest" way to reach out with little consequence was with a ping command)